### ISSUE TITLE
bap functions with multiple entry points and/or disjoint connected components

### ISSUE THREAD

**Issue by dluchaup** [BinaryAnalysisPlatform/bap#325]
In some cases the graph for the blocks in a function has multiple roots and/or disjoint connected components. (See also issue #324)
Using the attached plugin and test executable, we get
```
 bap test_checked_malloc0.O3.x --use-ida --no-byteweight -l sub_check --loader=bap-elf 
 program:test_checked_malloc0.O3.x sub:main #roots=2
```
My understanding of bap symbols is that currently they can only represent functions.
Additionally, the attached plugin reports 111 functions with multiple roots on a private test suite.

[ex.zip](https://github.com/BinaryAnalysisPlatform/bap/files/89006/ex.zip)

**Comment by ivg** [comment]
Please, provide me the test suite. 
And by "root" you mean a block that doesn't have predecessors?

**Comment by dluchaup** [comment]
Yes, 'root' in this case means it doesn't have a predecessor. 
My understanding is that the first block in a function (or the first node in the graph of a function) is the "official" entry point. In most cases this should have 0 predecessors (exceptions are possible for tail recursion).
I'll send you a private email about the test suite results.

**Comment by ivg** [comment]
should be fixed in #344 
if still persists, feel free to reopen
